pact/Pact/Core/IR/Eval/CoreBuiltin.hs

```diff
@@ -907,15 +936,18 @@ coreReadDecimal :: (CEKEval step b i m, MonadEval b i m) => NativeFunction step
 coreReadDecimal info b cont handler _env = \case
   [VString s] -> do
     viewEvalEnv eeMsgBody >>= \case
-      PObject envData ->
+      PObject envData -> do
+        chargeGasArgs info $ GObjOp $ ObjOpLookup s $ M.size envData
         case M.lookup (Field s) envData of
           Just (PDecimal p) -> returnCEKValue cont handler (VDecimal p)
           -- See [Note: Parsed Decimal]
           Just (PInteger i) -> returnCEKValue cont handler (VDecimal (Decimal 0 i))
-          Just (PString raw) -> case parseNumLiteral raw of
-            Just (LInteger i) -> returnCEKValue cont handler (VDecimal (Decimal 0 i))
-            Just (LDecimal l) -> returnCEKValue cont handler (VDecimal l)
-            _ -> returnCEK cont handler (VError "read-decimal failure" info)
+          Just (PString raw) -> do
+            chargeGasArgs info $ GStrOp $ StrOpLength $ T.length raw
```

Comment:
Should this not charge on `StrOpParse`?

Reply:
In fact, it's even better than that, it should be `StrOpConvToInt` which has precisely the required factor! Thanks for spotting this, addressed.